lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -27,6 +27,13 @@ class CyberSourceGateway < Gateway
       # Schema files can be found here: https://ics2ws.ic3.com/commerce/1.x/transactionProcessor/
       TEST_XSD_VERSION = '1.164'
       PRODUCTION_XSD_VERSION = '1.164'
+      ECI_BRAND_MAPPING = {
```

Comment:
There are more values in [app_commerce_indicators](https://developer.cybersource.com/library/documentation/dev_guides/CC_Svcs_SO_API/html/Topics/app_commerce_indicators.htm) than in `ECI_BRAND_MAPPING`. Is it because you are only mapping the non 3-DS cases (ie `vbv` but not `vbv_attempted` or `vbv_failure`), because the `eci` is passed in `options[:three_d_secure]` in the 3DS cases?

Reply:
it looks like cybersource has `eciRaw` and `commerceIndicator` as separate variables and they treat `eciRaw` as 3DS specific?

Reply:
Yes, because we don't send any 3ds value in case of attempt or failure, do we?

Reply:
Shopify does not attempt the payment if the buyer failed the challenge. In the other cases, Shopify provides 3DS data in case the 3DS authentication was attempted, even if it errorred.

Reply:
How could we know that the authentication was attempted and not successful? Are there anything from the three_d_secure option that would hint to that?

Reply:
For example, in the Cardinal [Test Case 13: Error on Authentication](https://cardinaldocs.atlassian.net/wiki/spaces/CCen/pages/903577725/EMV+3DS+2.0+Test+Cases#EMV3DS2.0TestCases-TestCase13:ErroronAuthentication)


ECI Value 07 | Visa, American Express, Discover (Diners Club), CB Visa, ELO
-- | --
ECI Value 00 | Mastercard, CB Mastercard

Reply:
Should I append an `attempted` to the commerceIndicator when the eci is 00 or 07 then? Seems like it would make sense. What do you think?

Reply:
I do not know without reading cybersource doc more.
We already provide the `eci`
```
xml.tag!('eciRaw', threeds_2_options[:eci]) if threeds_2_options[:eci]
```
so I do not understand yet why we have to also provide `commerceIndicator`

Reply:
Yeah... me neither... anyway, I'll merge that for now and talk with them a bit more, this is still a good change to merge.

Reply:
If it helps, `commerceIndicator` is also utilized for specifying certain COF transaction types, i.e. `recurring` `install` etc.